### PR TITLE
[go_router] Add `params` and `queryParams` to `.go()`

### PR DESCRIPTION
**Pull request by ValentinVignal**
Resolves https://github.com/flutter/flutter/issues/108173

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/go_router/lib/src/router.dart [hunk 1]
```diff
@@ -133,13 +134,32 @@ class GoRouter extends ChangeNotifier with NavigatorObserver {
...
+    if (params.isNotEmpty || queryParams.isNotEmpty) {
+      parsedLocation = parsePathWithParameters(
+        location,
+        params: params,
+        queryParams: queryParams,
+      );
+    } else {
+      parsedLocation = location;
+    }
     _routeInformationProvider.value =
```
Comment (ValentinVignal) [inline packages/go_router/lib/src/router.dart:161]: I had to do this to not make it a breaking change.

If it goes through `parsePathWithParameters` all the time and `location = '/family?fid=f2'` (for example), the location given to `RouteInformation` will be `'/family%3Ffid=f2'`

**Comment by chunhtai** [comment]
Let's discuss this in the issue first.

**Comment by ValentinVignal** [comment]
@chunhtai sure. Are you talking about this one? https://github.com/flutter/flutter/issues/108173 Because it got closed and I'm not sure I understand what is the conclusion

**Comment by chunhtai** [comment]
The conclusion is this API is probably not worth adding since people can use Uri class to format the string if they want.

**Comment by chunhtai** [comment]
closing this pr

**Pull request closed without merging**